motionplan/constraint.go

```diff
@@ -213,36 +226,40 @@ func NewCollisionConstraint(
 	}
 
 	// create constraint from reference collision graph
-	constraint := func(state *ik.State) bool {
+	constraint := func(state *ik.State) error {
 		var internalGeoms []spatial.Geometry
 		switch {
 		case state.Configuration != nil:
 			internal, err := state.Frame.Geometries(state.Configuration)
 			if err != nil {
-				return false
+				return err
 			}
 			internalGeoms = internal.Geometries()
 		case state.Position != nil:
-			// TODO(RSDK-5391): remove this case
 			// If we didn't pass a Configuration, but we do have a Position, then get the geometries at the zero state and
 			// transform them to the Position
 			internal, err := state.Frame.Geometries(make([]referenceframe.Input, len(state.Frame.DoF())))
 			if err != nil {
-				return false
+				return err
 			}
 			movedGeoms := internal.Geometries()
 			for _, geom := range movedGeoms {
 				internalGeoms = append(internalGeoms, geom.Transform(state.Position))
 			}
 		default:
-			return false
+			return errors.New("need either a Position or Configuration to be set for a ik.State")
 		}
 
 		cg, err := newCollisionGraph(internalGeoms, static, zeroCG, reportDistances, collisionBufferMM)
 		if err != nil {
-			return false
+			return err
+		}
+		cs := cg.collisions(collisionBufferMM)
+		if len(cs) != 0 {
+			// we could choose to amalgamate all the collisions into one error but its probably saner not to and choose just the first
```

Comment:
If we're not combining all, could we sort so that for a given collision scenario you always get the same pair back?

Reply:
This is going to run a fair number of times considering it will be hit every time an iteration is tested for any reason.  I thought about doing that but I dont think its worth the performance hit.  I also think the stochasticity is slightly good because it shows all the possibilities